src/v/CMakeLists.txt

```diff
@@ -37,23 +37,39 @@ add_subdirectory(compat)
 add_subdirectory(rp_util)
 
 option(ENABLE_GIT_VERSION "Build with Git metadata" OFF)
+# NOTE: if enabled, the git hash is included in the Redpanda binary, which will
+# require relinking of at least the `redpanda` binary each time the hash
+# changes (switching branches, creating a new commit, etc).
+option(ENABLE_GIT_HASH "Build with Git hash in metadata" OFF)
 if(${ENABLE_GIT_VERSION})
-include(GetGitRevisionDescription)
-get_git_head_revision(GIT_REFSPEC GIT_SHA1)
-git_local_changes(GIT_CLEAN_DIRTY)
-if("${GIT_CLEAN_DIRTY}" STREQUAL "DIRTY")
-  set(GIT_CLEAN_DIRTY "-dirty")
-else()
-  set(GIT_CLEAN_DIRTY "")
-endif()
-git_describe(GIT_VER --always)
+  include(GetGitRevisionDescription)
+  if (${ENABLE_GIT_HASH})
+    git_describe(GIT_VER --always)
+  else(${ENABLE_GIT_HASH})
+    # If displaying a version, make sure it also doesn't display a hash.
+    git_describe(GIT_VER --always --abbrev=0)
+  endif()
 else(${ENABLE_GIT_VERSION})
   set(GIT_VER "no_version")
+  # ENABLE_GIT_VERSION=off disables Git versioning altogether.
+  set(ENABLE_GIT_HASH OFF)
```

Comment:
Maybe better to assert above that `ENABLE_GIT_HASH` cannot be `ON` if `ENABLE_GIT_VERSION` is `OFF`

Reply:
or maybe move the calls to `set(GIT_SHA1)` && `set(GIT_CLEAN_DIRTY)` in here and just don't set the option.

Reply:
Done, opted for the first